frontend/src/pages/SignUp/index.tsx

```diff
@@ -98,69 +72,94 @@ const SignUp: React.FC = () => {
-            <CustomTextField
+            <TextField
               label="Username"
-              size="small"
               required
-              emptyField={emptyFields.username}
-              validator={usernameValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("username", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              {...register("username", { validate: { usernameValidator } })}
+              error={!!errors.username}
+              helperText={errors.username?.message}
             />
-            <CustomTextField
+            <TextField
               label="Email"
-              size="small"
               required
-              emptyField={emptyFields.email}
-              validator={emailValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("email", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              type="email"
+              {...register("email", { validate: { emailValidator } })}
+              error={!!errors.email}
+              helperText={errors.email?.message}
             />
-            <CustomTextField
+            <TextField
               label="Password"
-              size="small"
               required
-              emptyField={emptyFields.password}
-              validator={passwordValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("password", value, isValid)
-              }
-              isPasswordField
+              fullWidth
+              margin="normal"
+              type="password"
+              {...register("password", { validate: { passwordValidator } })}
+              error={!!errors.password}
+              helperText={errors.password?.message}
+              typeof={showPassword ? "text" : "password"}
+              slotProps={{
+                input: {
+                  endAdornment: (
+                    <InputAdornment position="end">
+                      <IconButton
+                        onClick={() => setShowPassword(!showPassword)}
+                        edge="end"
+                      >
+                        {showPassword ? (
+                          <VisibilityOff
+                            sx={(theme) => ({ fontSize: theme.spacing(2.5) })}
+                          />
+                        ) : (
+                          <Visibility
+                            sx={(theme) => ({ fontSize: theme.spacing(2.5) })}
+                          />
+                        )}
+                      </IconButton>
+                    </InputAdornment>
+                  ),
+                },
+              }}
             />
             <Button
               type="submit"
               variant="contained"
-              sx={(theme) => ({ height: theme.spacing(5) })}
+              sx={(theme) => ({ margin: theme.spacing(2, 0) })}
```

Comment:
If you're okay with having the` size="small"` attribute on the form fields (as mentioned above!), should we keep the style on the height rather than the margin? To keep the button's height consistent with the form fields